debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -450,8 +449,10 @@ private Struct schemaChangeRecordValue(SchemaChangeEvent event) {
         }
 
         @Override
-        public void schemaChangeEvent(SchemaChangeEvent event) throws InterruptedException {
-            historizedSchema.applySchemaChange(event);
+        public void schemaChangeEvent(HistorizedDatabaseSchema<T> historizedSchema, SchemaChangeEvent event) throws InterruptedException {
+            if (historizedSchema != null) {
```

Comment:
I'm not sure if this change was necessary, probably worth reverting. Can a schema change event happen on a non-historized schema?